tools/emr/submit_datagen_job.py

```diff
@@ -79,9 +81,15 @@ def submit_datagen_job(name, sf,
                        conf=None
                        ):
 
+    # for very large scale factors, the availability zone might run out of machines
+    if sf >= 10000:
+        instance_type = 'i3.4xlarge'
```

Comment:
this overrides user supplied values, so we shouldn't do this. we could refactor the function, to have the `None` value as default for both parameters, in which case will auto determine the parameters. You can write a small helper function to do this based of the SF.
```python
auto_instance_type, auto_sf_ratio = auto_instance_config(sf)
instance_type = instance_type || auto_instance_type
sf_ratio = auto_sf_ratio
```